mne/channels/channels.py

```diff
@@ -1536,3 +1539,150 @@ def make_1020_channel_selections(info, midline="z"):
                       for selection, picks in selections.items()}
 
     return selections
+
+
+def combine_channels(instance, groups, method='mean', keep_stim=False,
+                     drop_bad=False):
+    """Combine channels based on specified channel grouping.
+
+    Parameters
+    ----------
+    instance : instance of Raw, Epochs, or Evoked
+        An MNE-Python object to combine the channels for. The object can be of
+        type Raw, Epochs, or Evoked.
+    groups : dict
+        Specifies which channels are aggregated into a single channel, with
+        aggregation method determined by the ``method`` parameter. One new
+        pseudo-channel is made per dict entry; the dict values must be lists of
+        picks (integer indices of ``ch_names``). For example::
+
+            groups=dict(Left=[1, 2, 3, 4], Right=[5, 6, 7, 8])
+
+        Note that within a dict entry all channels must have the same type.
+    method : str | callable
+        Which method to use to combine channels. If a :class:`str`, must be one
+        of 'mean', 'median', or 'std' (standard deviation). If callable, the
+        callable must accept one positional input (data of shape ``(n_channels,
+        n_times)``, ``(n_epochs, n_channels, n_times)``, or ``(n_evokeds,
+        n_channels, n_time)``) and return an :class:`array <numpy.ndarray>` of
+        shape ``(n_times)``, ``(n_epochs, n_times)``, or ``(n_evokeds,
+        n_times)``. For example with an instance of Raw::
+
+            method = lambda data: np.median(data, axis=0)
+
+        Defaults to ``mean``.
+    keep_stim : bool
+        If True (default False), keep stimulus channels.
+    drop_bad : bool
+        If True (default False), drop channels marked as bad.
+
+    Returns
+    -------
+    combined_inst : instance of Raw, Epochs, or Evoked
+        An MNE-Python object with channels combined for each group defined in
+        the ``groups`` parameter.
+    """
+    from ..io import BaseRaw, RawArray
+    from ..epochs import BaseEpochs
+    from ..evoked import Evoked
```

Comment:
you could import these from `..` as well; they're exposed at the root level of the package. (unimportant; FYI)